### PR TITLE
Added support for Permissions API

### PR DESCRIPTION
**Pull request by JasonStoltz**
- Added support for Permissions API
- Added 'mock_endpoint' and 'fixtures' for easier testing - We are not using a tool like VCR, etc. for capturing and replaying responses as we do in other languages, so we need to manually wire this up ourselves.
- Used an auto formatter to better format files

### CODE REVIEW

File: tests/apis/fixtures/list_all_permissions_with_paging_response.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+list_all_permissions_with_paging_response = {
+    'meta': {
+        'page': {
+            'current': 2,
+            'total_results': 2,
+            'total_pages': 1,
+            'size': 20
+        }
+    },
+    'results': [{
```
Comment (yakhinvadim) [inline tests/apis/fixtures/list_all_permissions_with_paging_response.py]: Does API respond with non-empty results array if we're requesting a page 2? I got an empty array in this case: https://github.com/elastic/enterprise-search-node/blob/master/test/test.js#L168-L178

**Review by yakhinvadim — CHANGES_REQUESTED** [review]

File: tests/apis/fixtures/list_all_permissions_with_paging_response.py (hunk 1 above)
Comment (JasonStoltz) [reply tests/apis/fixtures/list_all_permissions_with_paging_response.py]: I mean, it just depends on the data set. If there's results on page 2, then yes.

Comment (JasonStoltz) [reply tests/apis/fixtures/list_all_permissions_with_paging_response.py]: I updated the counts so that they add up correctly

**Review by yakhinvadim — DISMISSED** [review]

**Comment by JasonStoltz** [comment]
@yakhinvadim Do you mind giving it another ✅ ? I broke the tests 😲

**Comment by JasonStoltz** [comment]
I'm going to use my administrator access to merge this PR.

**Pull request merged**